File: python/2_keywords/menu_item.py
# menu_item.py
from abc import ABC, abstractmethod

class MenuItem(ABC):
    BASE_PRICE = 2.0  # convention for constant

    def __init__(self, name: str):
        self.name = name  # 'self' like 'this'

    @abstractmethod
    def get_price(self) -> float:
        pass

    def display(self):
        print(f"{self.name} - ${self.get_price()}")

class Pastry(MenuItem):
    def __init__(self, name: str, type_: str):
        super().__init__(name)  # super ctor/method
        self.type = type_

    def get_price(self) -> float:
        return MenuItem.BASE_PRICE + 2.5

class Cafe:
    total_items_sold = 0  # class variable (static-like)

    @classmethod
    def sell(cls, item: MenuItem):
        item.display()
        cls.total_items_sold += 1

# Demo - no 'new', just call
croissant = Pastry("Croissant", "Butter")
Cafe.sell(croissant)
print("Total sold:", Cafe.total_items_sold)

File: typescript/2_keywords/menu-item.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
// menu-item.ts
var MenuItem = /** @class */ (function () {
    function MenuItem(name) {
        this.name = name;
        this.basePrice = 2.0; // constant-like
    }
    MenuItem.prototype.display = function () {
        console.log("".concat(this.name, " - $").concat(this.getPrice())); // 'this'
    };
    return MenuItem;
}());
var Pastry = /** @class */ (function (_super) {
    __extends(Pastry, _super);
    function Pastry(name, type) {
        var _this = _super.call(this, name) || this; // super ctor
        _this.type = type;
        return _this;
    }
    Pastry.prototype.getPrice = function () {
        return this.basePrice + 2.5;
    };
    return Pastry;
}(MenuItem));
var Cafe = /** @class */ (function () {
    function Cafe() {
    }
    Cafe.sell = function (item) {
        item.display();
        Cafe.totalItemsSold++;
    };
    Cafe.totalItemsSold = 0;
    return Cafe;
}());
// Demo
var croissant = new Pastry("Croissant", "Butter");
Cafe.sell(croissant);
console.log("Total sold:", Cafe.totalItemsSold);
